pandas/core/dtypes/cast.py

```diff
@@ -1523,6 +1523,28 @@ def find_common_type(types: List[DtypeObj]) -> DtypeObj:
     return np.find_common_type(types, [])
 
 
+def construct_2d_arraylike_from_scalar(
+    value: Scalar, length: int, width: int, dtype: np.dtype, copy: bool
```

Comment:
can u integrate with the 1d version and just require a ndim arg?

Reply:
they're pretty different; we could wrap both in a caller that has a shape arg